### PR TITLE
refactor(ninjafile): simplify ninja.kak and fix few things

### PR DESCRIPTION
**Pull request by tototest99**
I’m sorry for these multiples PR on ninjafiles rather than one [WIP]; but everytime I’m not sure I will have further time to work on it, so end up submitting them as better than nothing.

### CODE REVIEW

File: rc/filetype/ninja.kak [hunk 1]
```diff
@@ -40,41 +40,29 @@ add-highlighter shared/ninja regions
...
-add-highlighter shared/ninja/command region '^\h+command' '[^$]\n' group
-add-highlighter shared/ninja/command/command regex 'command' 0:variable
-add-highlighter shared/ninja/command/equal regex '=' 0:operator
-add-highlighter shared/ninja/command/linebreak regex '\$$' 0:operator
-add-highlighter shared/ninja/command/variables regex '\$\w+|\$\{\w+\}' 0:value
+# `subninja`, `include`, `rule`, `pool` and `default` declarations
+add-highlighter shared/ninja/sirpd region '^(subninja|include|rule|pool|default)\b' '[^$]\n' group
+add-highlighter shared/ninja/sirpd/default regex '^(subninja|include)\b' 0:module
+add-highlighter shared/ninja/sirpd/rulepool regex '^(rule|pool|default)\b' 0:keyword
+add-highlighter shared/ninja/sirpd/linebreak regex '\$$' 0:operator
```
Comment (lenormf) [inline rc/filetype/ninja.kak:47]: Those lines could be factorised into one, and remain fairly readable.

File: rc/filetype/ninja.kak [hunk 2]
```diff
@@ -40,41 +40,29 @@ add-highlighter shared/ninja regions
...
 
 # `build`
-add-highlighter shared/ninja/build region '^build' '\n' group
-add-highlighter shared/ninja/build/build regex '^build\h+' 0:keyword
-add-highlighter shared/ninja/build/rule regex ':\h+(\w+)' 0:function
+add-highlighter shared/ninja/build region '^build\b' '[^$]\n' group
+add-highlighter shared/ninja/build/build regex '^build\b' 0:keyword
+add-highlighter shared/ninja/build/rule regex ':\h+((\w|-)+)' 0:function
```
Comment (lenormf) [inline rc/filetype/ninja.kak:52]: Ditto.

File: rc/filetype/ninja.kak [hunk 3]
```diff
@@ -83,10 +71,11 @@ evaluate-commands %sh{
 
   reserved_names_regex="$(echo ${reserved_names} | tr ' ' '|')"
   printf %s "
-      add-highlighter shared/ninja/variable/reserved_names regex ${reserved_names_regex} 0:meta
+      add-highlighter shared/ninja/variable/reserved_names regex '^\h*(${reserved_names_regex})' 0:meta
```
Comment (lenormf) [inline rc/filetype/ninja.kak:74]:
```suggestion
      add-highlighter shared/ninja/variable/reserved_names regex '^\h*(${reserved_names_regex})\b' 0:meta
```

File: rc/filetype/ninja.kak [hunk 4]
```diff
@@ -103,8 +92,8 @@ define-command -hidden ninja-indent-on-new-line %{
         try %{ execute-keys -draft \; K <a-&> }
         # filter previous line
         try %{ execute-keys -draft k : ninja-trim-indent <ret> }
-        # indent after lines begining with rule and pool (do people want build too ?)
-        try %{ execute-keys -draft \; k x <a-k> ^(\brule|pool) <ret> j <a-gt> }
+        # indent after lines begining with rule and pool
+        try %{ execute-keys -draft \; k x <a-k> ^(\brule|pool|build) <ret> j <a-gt> }
```
Comment (lenormf) [inline rc/filetype/ninja.kak:96]:
```suggestion
        try %{ execute-keys -draft \; k x <a-k> ^(rule|pool|build)\b <ret> j <a-gt> }
```

File: rc/filetype/ninja.kak (hunk 1 above)
Comment (tototest99) [reply rc/filetype/ninja.kak:47]: I’m afraid I’m out of time for now, feel free to take over or suggest and I will validate.

**Comment by mawww** [comment]
Could you rebase this on latest master please ?

**Pull request merged**